### ISSUE TITLE
Upgrades should feel like kitting out a ship, instead of equiping an escalating amount of items

### ISSUE THREAD

**Issue by EchelonFour** [JamesHDuffield/ascii-rust#11]
I think doing an FTL style thing where there are systems and slots and power diverted to slots and whatnot would be more on theme.
I'm not expecting a full ship LCARS UI, I think a list of each ship system and a bar next to it inidicating how leveled it is and a plus button would do it.

Down the line, you could limit the starting systems to only the basics and change the pickups to two classes: "XP" and "Item" to allow you to install new ones.

And further down the line, I reckon you allow the player to pause (by holding space or something) to reassign power between the systems. Give the player a "full power to forward laser" moment.

**Comment by JamesHDuffield** [comment]
Yeah this is really just a basic Vampire Survivors clone to get a feel for what things are fun.

I like the idea of ship slots and then equipment rather than pure Turret + Passives with a visual representation of your load out on level up.

You can currently pause "Escape" key by the way, but not sure I want micromanaging during combat to be a big part of it.

There is the potential to move from a single location with infinite spawning, to a clear out area, collect loot then warp to next area flow instead.

**Comment by EchelonFour** [comment]
I thought about the micro managing thing. If you limit changes to every X game seconds (or an upgrade), then you make it not be a boulder's gate RPG, but you have the option to change your mind a little.
X feels like it should be 10 or 30 or something.

**Comment by JamesHDuffield** [comment]
Let's get some playtesters to have a go and find what parts of the design work or don't work, before re-assessing the loop.
